sanic_ext/extensions/openapi/openapi.py

```diff
@@ -178,24 +178,32 @@ async def handler(request, *handler_args, **handler_kwargs):
 
 def parameter(
     name: Optional[str] = None,
-    schema: Type = str,
-    location: str = "query",
+    schema: Optional[Type] = None,
+    location: Optional[str] = None,
     parameter: Optional[Parameter] = None,
     **kwargs,
 ):
     if parameter:
-        if name:
+        if name or schema or location:
             raise SanicException(
                 "When using a parameter object, you cannot pass "
                 "other arguments."
             )
-
-        name = parameter.fields["name"]
-        schema = parameter.fields["schema"]
-        location = parameter.fields["in"]
+    if not schema:
+        schema = str
+    if not location:
+        location = "query"
 
     def inner(func):
-        OperationStore()[func].parameter(name, schema, location, **kwargs)
+        if parameter:
+            # Temporary solution convert in to location,
+            # need to be changed later.
+            fields = dict(parameter.fields)
```

Comment:
This is a walk-around solution, do you think it is acceptable for this PR or needs a further completed solution for the whole API?

Reply:
I think it is okay for now. Next year we need a good overhaul to this API. Let's not worry about fixing it just yet.

Reply:
Thanks.